src/trusted/service_runtime/nacl_syscall_common.c

```diff
@@ -897,7 +895,8 @@ int32_t NaClSysRead(struct NaClAppThread  *natp,
     goto out;
   }
 
-  ndp = NaClGetDesc(nap, fd);
+  ndp = (struct NaClDesc *) DynArrayGet(&nap->desc_tbl, fd);
+  /* Use DynArrayGet to bypass GetDesc locks for fast IO */
```

Comment:
I also think this is supposed to add a new ref to prevent the desc from being freed in between, this may or may not be something we need to worry about.